### PR TITLE
AirDC++ Socket Status 

### PR DESCRIPTION
**Pull request by rishighan**
This PR covers work to achieve the following:

1. A navbar icon that depicts in real-time, the connection to the AirDC++ web client via the `airdcpp-websocket-api`
2. Surfacing session information for a `connected` status
3. Surfacing error information for a `disconnected` status

### CODE REVIEW

File: src/client/actions/airdcpp.actions.tsx [hunk 1]
```diff
@@ -32,180 +37,206 @@ function sleep(ms: number): Promise<NodeJS.Timeout> {
...
+        break;
+    }
+  };
+export const search =
+  (data: SearchData, ADCPPSocket: any, credentials: any) =>
+  async (dispatch) => {
+    try {
+      if (!ADCPPSocket.isConnected()) {
+        await ADCPPSocket.connect(
+          credentials.username,
```
Comment (maksis) [inline src/client/actions/airdcpp.actions.tsx]: Shouldn't you just call `connect` without any arguments since everything is set in the socket options?

**Comment by nstephenh** [comment]
That's definitely react code :P
I don't see anything that seems wrong at a cursory glance. Lots of stuff I'm not familiar with, much of it project specific. Hopefully I can contribute more after looking at a few more of these/the project as a whole.

File: src/client/actions/airdcpp.actions.tsx (hunk 1 above)
Comment (rishighan) [reply src/client/actions/airdcpp.actions.tsx]: Yeah, I never tried testing it without params, after I implemented the change you suggested @maksis. I'll remove the params and push.

Comment (rishighan) [reply src/client/actions/airdcpp.actions.tsx]: @maksis Addressed in https://github.com/rishighan/threetwo/pull/58/commits/3102b489c8db55594a79ba84a9789dc9cd65493f

Comment (maksis) [reply src/client/actions/airdcpp.actions.tsx]: > @maksis Addressed in [3102b48](https://github.com/rishighan/threetwo/commit/3102b489c8db55594a79ba84a9789dc9cd65493f)

But the same code is still everywhere...

Comment (rishighan) [reply src/client/actions/airdcpp.actions.tsx]: Grrrr... jk. Yes I'll update that

Comment (rishighan) [reply src/client/actions/airdcpp.actions.tsx]: Aaaand done: https://github.com/rishighan/threetwo/pull/58/commits/1d8dd3ca3e6d9b0b4a3cb4f6ba462a3de19daacf

**Review by maksis — APPROVED** [review]

**Pull request merged**